Test/EncapsulationTestAwareTrait.php

```diff
@@ -0,0 +1,277 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchBundle\Test;
+
+/**
+ * A helper for testing setters/getters.
+ *
+ * Trait EncapsulationTestAwareTrait
+ */
+trait EncapsulationTestAwareTrait
```

Comment:
There might be scenarios, where we need to test encapsulation, but we're already using abstraction class.